java/vector/src/main/java/org/apache/arrow/vector/NullableIntVector.java

```diff
@@ -0,0 +1,122 @@
+package org.apache.arrow.vector;
+
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.vector.complex.impl.IntReaderImpl;
+import org.apache.arrow.vector.complex.reader.FieldReader;
+import org.apache.arrow.vector.holders.IntHolder;
+import org.apache.arrow.vector.holders.NullableIntHolder;
+import org.apache.arrow.vector.types.Types;
+import org.apache.arrow.vector.types.pojo.FieldType;
+
+public class NullableIntVector extends BaseNullableFixedWidthVector {
+   private static final int TYPE_WIDTH = 4;
+   private final FieldReader reader;
+
+   public NullableIntVector(String name, BufferAllocator allocator) {
+      this(name, FieldType.nullable(org.apache.arrow.vector.types.Types.MinorType.INT.getType()),
+              allocator);
+   }
+
+   public NullableIntVector(String name, FieldType fieldType, BufferAllocator allocator) {
+      super(name, allocator, fieldType, TYPE_WIDTH);
+      reader = new IntReaderImpl(NullableIntVector.this);
+   }
+
+   @Override
+   public FieldReader getReader(){
+      return reader;
+   }
+
+   @Override
+   public Types.MinorType getMinorType() {
```

Comment:
this might be a good time to remove the MinorTypes and just use FieldTypes - I believe @jacques-n discussed this a while ago, that they don't actually capture all the field information.

Reply:
If we could add a `getType` to return the `ArrowType` obj, that would be great.  Something like what #972 proposed to avoid having to cast would be helpful. But I'd hate to add more work to this, maybe better to discuss after the main stuff is done.

Reply:
I also found `MinorType` to be weird. But it's probably hard to remove due to compability? We could consider deprecating those though.

Reply:
Added to requirement 7 in the requirement doc. Let's keep this out of this PR and discuss separately?

Reply:
Yes, best to discuss separately. We will be merging multiple multiple patches to the new branch "java vector refactor" . We can do this once rest of the new code is stabilized, existing functionality is not broken etc. 

Later on we can merge this branch into master once refactoring is done or this particular thing can be addressed directly on master once refactor branch has been merged. I am open to ideas.